Using ~0 instead of -1 as magic number for end().

rolanddenis opened the pull request:
Using -1 as magic number for unsigned int is implementation dependant.
~0 is a better choice and is type-size independant.

dcoeurjo commented:
Thanks. It was not in your PR but could you please uncomment the CONCEPT_ASSERT in Labels.h  (and check what was wrong) ?

dcoeurjo commented:
(please have a look to my comment on the previous PR)

rolanddenis commented:
Concerning the CONCEPT_ASSERT, the concept name was wrong. I corrected it (to CUnsignedNumber) and added CIntegralNumber concept check. I also added a test block in testLabels.cpp related to the iterator bug.
Could you please wait to merge this PR: I have another bug with LabelledMap that I wish to include in this PR.

dcoeurjo commented:
Great. thanks. 
Merging
